mne/time_frequency/tests/test_psd.py

```diff
@@ -160,6 +160,116 @@ def test_psd():
         assert (psds_ev.shape == (len(kws['picks']), len(freqs)))
 
 
+def test_psd_welch_average_kwarg_raw():
+    """Test `average` kwarg of psd_welch() with Raw input data."""
+    raw = read_raw_fif(raw_fname)
+    picks_psd = [0, 1]
+
+    # Populate raw with sinusoids
+    rng = np.random.RandomState(40)
+    data = 0.1 * rng.randn(len(raw.ch_names), raw.n_times)
+    freqs_sig = [8., 50.]
+    for ix, freq in zip(picks_psd, freqs_sig):
+        data[ix, :] += 2 * np.sin(np.pi * 2. * freq * raw.times)
+
+    raw = RawArray(data, raw.info)
+
+    tmin, tmax = 0, 20  # use a few seconds of data
```

Comment:
how long does it take to test? tests should be as fast as possible

Reply:
I've changed this to `tmin, tmax = -0.5, 0.5`; the entire test (i.e. sum of runs for all three parametrizations `raw`, `epochs`, `evoked`) now finishes in less than 800 ms on my laptop computer.